test/e2e/flowaggregator_test.go

```diff
@@ -115,14 +116,15 @@ const (
 	egressAntreaNetworkPolicyName  = "test-flow-aggregator-antrea-networkpolicy-egress"
 	testIngressRuleName            = "test-ingress-rule-name"
 	testEgressRuleName             = "test-egress-rule-name"
-	collectorCheckTimeout          = 12 * time.Second
 	iperfTimeSec                   = 12
 	// Single iperf run results in two connections with separate ports (control connection and actual data connection).
 	// As 2s is the export active timeout of flow exporter and iperf traffic runs for 12s, we expect totally 12 records
 	// exporting to the flow aggregator at time 2s, 4s, 6s, 8s, 10s, and 12s after iperf traffic begins.
 	// Since flow aggregator will aggregate records based on 5-tuple connection key and active timeout is 3.5 seconds,
 	// we expect 3 records at time 5.5s, 9s, and 12.5s after iperf traffic begins.
 	expectedNumDataRecords = 3
+	// serving as the external server to test records of pod-to-external flow type
+	externalServer = "google.com"
```

Comment:
out-of-curiosity, if we had chosen the gateway's IP, or at least the Node's public IP, wouldn't we have generated Pod-to-External records, without the need to connect to the Internet?

Reply:
gateway's IP:
Sorry I didn't figure out how to get the gateway's IP. I searched through the repo, knowing it can be accessed through `nodeConfig`, but don't know how to access `nodeConfig` here. So I used `gwIP, _, _ := net.ParseCIDR(“10.0.1.1/24”)`, not sure which is the right way. 
The behavior is:
```
flowaggregator_test.go:576: PING 10.0.1.1 (10.0.1.1): 56 data bytes
    
    --- 10.0.1.1 ping statistics ---
    5 packets transmitted, 0 packets received, 100% packet loss
```

Node's public IP:
I found the node's external ip is `None` on my local cluster. Also not sure it is expected or not.

Reply:
Internal IP is fine. You should be able to use this function and pass it the NodeSpec `https://github.com/antrea-io/antrea/blob/4181bd975901ecd4e694308db80b2494b9d792c6/pkg/util/k8s/node.go#L29`

It's probably a better idea than using the gateway's IP as:
* there is no gateway IP in networkPolicyOnly mode (although I don't think we ever run e2e tests in that mode)
* maybe we will not always classify Pod-to-Gateway flows as Pod-to-External

If you run an nginx server (or agnhost server) as a hostNetwork Pod, you should be able to reach it using the Node's IP. This will work for IPv6-only clusters as well, as Quan pointed out below.